rosetta-source/src/main/rosetta/margin-schedule-func.rosetta

```diff
@@ -694,277 +626,264 @@ func StandardizedScheduleMonetaryNotionalCurrencyFromResolvablePQ: <"Extracts th
         notionalCurrency string (0..1)
 
     set notionalCurrency:
-        if priceQuantity -> quantitySchedule exists
-                and priceQuantity -> quantitySchedule -> unit -> currency exists
-        then (if priceQuantity -> quantitySchedule -> multiplier exists
-            then priceQuantity -> quantitySchedule -> multiplier -> unit -> currency
-            else priceQuantity -> quantitySchedule -> unit -> currency)
 
-func FXFarLeg: <"Extracts the far leg of an FX swap (deliverable or not) based on two criteria: the forward payout with the latest value date or the forward payout with the latest settlement date.">
-    inputs:
-        product NonTransferableProduct (1..1)
-    output:
-        farLegPayout SettlementPayout (0..1)
+    if priceQuantity -> quantitySchedule exists and priceQuantity -> quantitySchedule -> unit -> currency exists
+            then (
+                if priceQuantity -> quantitySchedule -> multiplier exists
+                then priceQuantity -> quantitySchedule -> multiplier -> unit -> currency
+                else priceQuantity -> quantitySchedule -> unit -> currency
+            )
 
+//Potential move to other namespaces
+func FXFarLeg: <"Extracts the far leg of an FX swap (deliverable or not) based on two criteria: the forward payout with the latest value date or the forward payout with the latest settlement date.">
+    inputs: product NonTransferableProduct (1..1)
+    output: nearLegPayout SettlementPayout (0..1)
     alias settlementPayout: product -> economicTerms -> payout -> SettlementPayout
+    set nearLegPayout:
+        //value date
+        if (Qualify_ForeignExchange_Swap(product -> economicTerms ) or Qualify_ForeignExchange_NDS(product -> economicTerms ) ) and settlementPayout -> settlementTerms -> settlementDate -> valueDate exists then settlementPayout
+            max [ item -> settlementTerms -> settlementDate -> valueDate]
+        //adjustable or relative date
+        else if (Qualify_ForeignExchange_Swap(product -> economicTerms ) or Qualify_ForeignExchange_NDS(product -> economicTerms ) ) and settlementPayout -> settlementTerms -> settlementDate -> adjustableOrRelativeDate exists then settlementPayout
+            max [ AdjustableOrAdjustedOrRelativeDateResolution( item -> settlementTerms -> settlementDate -> adjustableOrRelativeDate)]
+
+func StandardizedScheduleFXSwapNotional: <"Extracts the notional amount and currency of an FX swap.">
+    inputs: 
+        farLeg SettlementPayout (0..1)
+        tradeLot TradeLot (0..1)
+    output: quantity NonNegativeQuantitySchedule (0..1)
+    alias priceQuantity:
+        tradeLot -> priceQuantity filter (item -> price = farLeg -> fixedPrice) then only-element
```

Comment:
due to the last change on version dev.84, should change the part 'farLeg -> fixedPrice' for 'farLeg -> priceQuantity -> priceSchedule'

Reply:
@manucarreramoreno In this case, can you please make this adjustment in your workspace and resubmit - it should update your PR (please don't make the change in GitHub directly).